libs/blocks/global-navigation/utilities.js

```diff
@@ -35,3 +38,71 @@ export function getLocale() {
 export function getCountry() {
   return getLocale()?.split('-').pop() || 'US';
 }
+
+// TODO this is just prototyped and should also support ?milolibs=test-branch
+const getOrigin = () => {
+  if (window.location.origin.includes('localhost') || window.location.origin.includes('.hlx.')) {
+    return window.location.origin;
+  }
+  return 'https://adobe.com';
+};
+
+async function fetchData(path) {
+  const resp = await fetch(path).catch(() => ({}));
+  const { data } = resp.ok ? await resp.json() : { data: [] };
+  const res = {};
+  for (let i = 0; i < data.length; i += 1) {
+    res[data[i].key] = data[i];
+  }
+  return res;
+}
+
+let fetchedPlaceholders;
+async function fetchPlaceholders() {
+  fetchedPlaceholders = fetchedPlaceholders || new Promise((resolve) => {
+    const { locale } = getConfig();
+    fetchData(`${getOrigin()}/libs/feds${locale.prefix}/placeholders.json`)
+      .then((res) => resolve(res));
+  });
+  return fetchedPlaceholders;
+}
+
+let fetchedDefaultPlaceholders;
+async function fetchDefaultPlaceholders() {
+  fetchedDefaultPlaceholders = fetchedDefaultPlaceholders || new Promise((resolve) => {
+    fetchData(`${getOrigin()}/libs/feds/placeholders.json`)
+      .then((res) => resolve(res));
+  });
+  return fetchedDefaultPlaceholders;
+}
+
+const keyToStr = (key) => key.replaceAll('-', ' ');
+
+export async function getPlaceholder(key = '') {
+  if (typeof key !== 'string') return { value: '' };
+  const { locale } = getConfig();
+  const defaultLocale = 'en-US';
+  const placeholders = await fetchPlaceholders();
+  if (placeholders[key]) return placeholders[key];
```

Comment:
If the key is not valid, `{ value: '' }` is returned. If the key exists in the fetched placeholders, it looks like you're returning the value directly, not `{ value: 'string for placeholder' }`. Is this the case? Am I missing something from the data structure?

Reply:
If it's not valid we return `{value: 'sign in'}` transforming the key (`sign-in`) 

If it is defined, we map whatever franklin delivers to us, so that this response
```json
{
 "total":5,
 "offset":0,
 "limit":5,
 "data":[{"key":"search","value":"Suchen"},{"key":"clear-results","value":"Ergebnisse löschen"},{"key":"try-advanced-search","value":"Erweiterte Suche"},{"key":"sign-in","value":"Anmelden", "link": "https://adobe.com/sign-in", analytics: "authored-analytics-label"},{"key":"view-account","value":"Account"}],
 ":type":"sheet"
}
```
turns to
```js
const placeholders = {
    "search": {
        "key": "search",
        "value": "Suchen"
    },
    "clear-results": {
        "key": "clear-results",
        "value": "Ergebnisse löschen"
    },
    "try-advanced-search": {
        "key": "try-advanced-search",
        "value": "Erweiterte Suche"
    },
    "sign-in": {
        "key": "sign-in",
        "value": "Anmelden",
        "analytics": "authored-analytics-label",
        "link": "https://adobe.com/sign-in"
    },
    "view-account": {
        "key": "view-account",
        "value": "Account"
    }
}
```
So `placeholders["sign-in"]` would be reference an object, so `{key: "sign-in", value: "Anmelden", ...rest}`

Originally I did that to retain the option to add more attributes, so if the placeholder sheet also contained a "link" column, or whatever attribute we want to pass, might be analytics related or any string, `placeholders["sign-in"]` would be `{key: "sign-in", value: "Anmelden", link: "https://adobe.com/sign-in", analytics: "special-authored-attribute"` instead of just referencing a string.

We don't use links now, but I don't think it hurts to stick to an object based structure so we can adapt to future changes more easily

Reply:
Just to recap and make sure I understand:
`placeholders["sign-in"]` will return:
1. `{key: "sign-in", value: "Something"}` if the key is found
2. `{key: "sign-in", value: "sign in"}` if the key is not found

`placeholders["sign in"]` will return:
3. `{value: ""}` since the key is not valid

Reply:
Yes, almost.. the gist it it should always return `value: ""` so u never need to worry about access to something undefined
`placeholders["sign-in"]` will return:
1. `{key: "sign-in", value: "Something-language-specific"}` if the key is found
2. `{key: "sign-in", value: "Something-english-backup"}` if the key was not found for a specific language, we try to fetch the English version
3. `{value: "sign in"}` if the key is not found

`placeholders["sign in"]` will return:
4. `{value: ""}` since the key is not valid